onnxruntime/core/framework/session_state.cc

```diff
@@ -43,9 +43,14 @@ const SequentialExecutionPlan* SessionState::GetExecutionPlan() const {
   return p_seq_exec_plan_.get();
 }
 
-void SessionState::AddInitializedTensor(int mlvalue_index, const MLValue& mlvalue) {
-  ORT_ENFORCE(mlvalue_index >= 0 && mlvalue_index <= mlvalue_name_idx_map_.MaxIdx());
-  initialized_tensors_.insert({mlvalue_index, mlvalue});
+Status SessionState::AddInitializedTensor(int mlvalue_index, const MLValue& mlvalue, const OrtDeleter& d) {
+  if (mlvalue_index < 0 || mlvalue_index > mlvalue_name_idx_map_.MaxIdx()) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "invalid mlvalue index ", mlvalue_index);
+  }
+  auto p = initialized_tensors_.insert({mlvalue_index, mlvalue});
+  if (!p.second) return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "duplicated mlvalue index ", mlvalue_index);
```

Comment:
>duplicated mlvalue index [](start = 72, length = 24)

nit: this is actually due to duplicate calls with the same name (meaning they get the same mlvalue index). an error message to that effect will be more helpful for a user.